search/searcher/search_numeric_range.go

```diff
@@ -53,20 +53,51 @@ func NewNumericRangeSearcher(indexReader index.IndexReader,
 	if !*inclusiveMax && maxInt64 != math.MinInt64 {
 		maxInt64--
 	}
+
+	var fieldDict index.AdvFieldDict
+	var err error
+	if irr, ok := indexReader.(index.IndexReaderRandom); ok {
+		fieldDict, err = irr.FieldDictRandom(field)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	var isIndexed filterFunc
+	if fieldDict != nil {
+		isIndexed = func(term []byte) bool {
+			if found, err := fieldDict.Exists(term); found && err == nil {
+				return true
+			}
+			return false
```

Comment:
minor thing -- (change not needed unless if there are other changes, imho)...

this might be done via a more compact... return err == nil && found